lib/posix/Kconfig

```diff
@@ -96,4 +96,12 @@ config	POSIX_MAX_OPEN_FILES
 endif
 endif # FILE_SYSTEM
 
+# The name of this option is mandated by zephyr_interface_library_named
+# cmake directive.
+config APP_LINK_WITH_PTHREAD
```

Comment:
should be APP_LINK_WITH_POSIX

Reply:
> should be APP_LINK_WITH_POSIX

No, as explained by @SebastianBoe, this should match the existing config symbols, that's why it's APP_LINK_WITH_PTHREAD.

Reply:
which existing config symbols? If anything is still using PTHREAD, change it POSIX as well.

Reply:
> If anything is still using PTHREAD, change it POSIX as well.

But this is not subject of this change. The subject of this change is "lib: posix: Switch to use zephyr_interface_library_named cmake directive".

Exploration of more deep changes to POSIX subsys was done in e.g. https://github.com/zephyrproject-rtos/zephyr/pull/12984 . Feel free to merge it if you like it.

Reply:
> But this is not subject of this change. The subject of this change is "lib: posix: Switch to use zephyr_interface_library_named cmake directive".

it becomes a subject of this change when you introduce a new Kconfig using the old name and we will be stuck with it.

Reply:
> we will be stuck with it.

Why would we be stuck with? We'll change it (and hundreds of other mistakes and bugs in Zephyr) when it comes to it.

All in all, I'll defer to @galak, who assigned me to work on POSIX subsys improvements, regarding whether that should be part of this PR or not. In the meantime, working on other assignment from him, and can't work further on this, beyond keeping the build green.